DeepSpeech.py

```diff
@@ -388,9 +384,11 @@ def train(server=None):
     train_set = DataSet(train_data,
                         FLAGS.train_batch_size,
                         limit=FLAGS.limit_train,
-                        next_index=lambda i: coord.get_next_index('train'))
+                        next_index=train_index.inc)
```

Comment:
nit: `next_index=lambda i: train_index += 1` and then we can avoid defining the `SampleIndex` class and just make this variable an integer.

Reply:
This is what I first tried. In Python `x += 1` is not resulting in a number and there is no non-hacky way to have multiple instructions in lambdas. If you go for an inline function instead, you've to make `train_index` global, as in Python non-local scope variables are only mutable if they are global. You'd have to do all of this twice for train and dev and it will get more verbose than the existing (cleaner) solution. This is Python...

Reply:
Ah, that's unfortunate. Alright!